### ISSUE TITLE
[ZoneJS] Angular 2 app hangs up on task queue

### ISSUE THREAD

**Issue by dragGH102** [angular/angular#9362]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report
```

I am posting this here because I think it's an issue related to Angular 2 even though I can't be completely sure as everything gets stuck when I try to debug it, so it could be System.Js as well as Zone.js.

**Current behavior**

The angular app (rc2 based on quickstart tutorial with a couple of other packages in map and packages (SystemJS config) hangs up, even blocking the browser completely (resulting in imposibility to debug)
I never had a similar issue with rc1 and related packages.

**Debug (what I could learn about he issue)**
- First of all, I think it's not a SystemJS issue since the following code (in index.html) prints 'app loaded):

        System.import('app')
                .then(function() { console.log('app loaded!'); })
                .catch(function(err){ throw(err); });

- Google Chrome just gets stuck (the UI is partially rendered)

- Mozilla Firefox (latest version) detects a script taking too much time and propose to debug it:
http://localhost:3000/node_mod…system.src.js line 1510 > eval:10248 # (0, eval)(getSource(load));

clicking on it I'm presented with zonejs instead and letting it loop (breakpoint -> continue -> hit same breakpoint again and so on), I see it keep going back to drainMicroTaskQueue() <-- within the first while(...) the loops eems to be infinitive.

As a side note also this is printed to console (I read the document but it's not very clear to me what the proposed change is):

```
*It looks like you're using the old forms module. This will be opt-in in the next RC, and
      will eventually be removed in favor of the new forms module. For more information, see:
      https://docs.google.com/document/u/1/d/1RIezQqE4aEhBRmArIAS1mRIZtWFf6JxN_7B4meyWK0Y/pub
```
 
* **Browser:** **all**
 
* **Language:** TypeScript (latest)

**Comment by zoechi** [comment]
What are the exact instructions to reproduce? Plunker?

**Comment by dragGH102** [comment]
@zoechi that's not possible. this is a commercial app.
I understand that without a plunkr it's difficult to say what's going. I wish I knew how to debug the issue (any suggestion is welcome) since the browser keeps getting stuck.
I'm also quite sure this is not an issue with the app's code as I already dealt with code running indefinitively but still the browser was usable.

**Comment by zoechi** [comment]
I wouldn't expect anybody to try to debug a whole application anyway. You would need to remove everything that's not related to the issue. 

The message about the old forms module you got, doesn't require any action. The forms module is going to be replaced soon. This is just a reminder that you're still using the old one. The old forms module will stay available for a while as forms-deprecated so you have time to migrate.

**Comment by dragGH102** [comment]
@zoechi I'll try to create a replicable example

**Comment by vicb** [comment]
Please re-open when you have a minimal repro in a plunker. Thanks.

**Comment by dragGH102** [comment]
I don't have a pluckr yet (feel free to close it again if necessary), but I probably found what the issue is (after that, it worked) :

A service (injected as provider in several views) was calling routing functions via 

        routerInstance. navigate(...) 

After removing it from the service and using it directly in the views (even though I believe routing can be useful in services, e. g. for **authentication** functions), the issue was solved. 

I didn't have this issue with Angular rc1 so it's apparently related to the fixes / features added in rc2 (but no breaking changes were listed). 

Hope this helps to detect a potential issue!

**Comment by dragGH102** [comment]
Actually I can't reopen it myself.

**Comment by pkozlowski-opensource** [comment]
@dragGH102 I'm sorry but the description doesn't help us much. What we need is a minimal reproduce scenario in plunker so we can actually _see_ the issue.

**Comment by sumanballapuramgh** [comment]
Hello,
As i have also experienced the same issue when i upgrade my project to rc2. Bunch of 
404 GET /node_modules/systemjs/dist/<various files>.js.map

**Comment by lassche** [comment]
For what it's worth, I've got a problem with the same symptoms: the browser (chrome) locks up 'after a while'. I'll see if I can upload my project later this week if that helps.

**Comment by dragGH102** [comment]
On Sat, Jun 25, 2016 at 8:18 PM, lassche <notifications@github.com> wrote:

> For what it's worth, I've got a problem with the same symptoms: the
> browser (chrome) locks up 'after a while'. I'll see if I can upload my
> project later this week if that helps.


That would be awesome!
I'm really concerned about this issue because it can't happen in a
production/commercial app (on which I'm working)

**Comment by Xardaska** [comment]
Hello,

we stumbled upon this problem while moving our legacy js api to Angular 2. 
It seems to have something to do with the `isThenable(value)` function of Zone.js cause it creates false positives (when used in conjunction with jQuery and ajax responses) which lead to some kind of endless loop (between `makeResolver(promise, state)` and `resolvePromise(promise, state, value)`).
I created a plunker which reproduces the problem. 
https://run.plnkr.co/plunks/pxGp3otI0wuVV0WGiywA/ 
The error handler of 
` testService.someCall().then(function() {
          console.log("Success");
        }, function() {
          console.log("Error");
        });`
will therefore never be called.
